src/vhpi/vhpi-model.c

```diff
@@ -574,6 +638,85 @@ static void init_range(c_range *r, tree_t t)
    r->IsUp = (rkind == RANGE_TO);
 }
 
+static void init_expr(c_expr *e, tree_t t, c_typeDecl *Type)
+{
+   e->Type = Type;
+
+   tree_flags_t flags = tree_flags(t);
+   if (flags & TREE_F_GLOBALLY_STATIC)
+      e->Staticness = vhpiGloballyStatic;
+   else if (flags & TREE_F_LOCALLY_STATIC)
+      e->Staticness = vhpiLocallyStatic;
+   else
+      e->Staticness = vhpiDynamic;
+}
+
+static void init_name(c_name *n, tree_t t, c_typeDecl *Type,
+                      vhpiStringT Name, vhpiStringT FullName)
+{
+   init_expr(&(n->expr), t, Type);
+   n->Name = n->CaseName = Name;
+   n->FullName = n->FullCaseName = FullName;
+}
+
+static void init_prefixedName(c_prefixedName *pn, c_typeDecl *Type,
+                              c_objDecl *simpleName, c_name *prefix,
+                              const char *suffix)
+{
+   vhpiStringT Name, FullName;
+
+   if (prefix != NULL) {
+      pn->Prefix = prefix;
+      Name = prefix->Name;
+      FullName = prefix->FullName;
+   } else {
+      Name = simpleName->decl.Name;
+      FullName = simpleName->decl.FullName;
+   }
+
+   vhpiStringT name = (vhpiStringT)xasprintf("%s%s", Name, suffix);
+   vhpiStringT fullname = (vhpiStringT)xasprintf("%s%s", FullName, suffix);
```

Comment:
Can we at least initialise the names lazily if not the whole object? Presumably these won't often be accessed. I think we could just initialise them to NULL and then create the string on the first use.

Reply:
Yes that would be preferable.